### ISSUE TITLE
Return and track drift integration metadata

### ISSUE THREAD

**Issue by n-west** [n-west/bliss#4]
For passing info in to drift integration it would be nice to have an option that passes drift rate ranges in physical units Hz/sec but still be able to pass discrete bins

some info that needs to come out of the drift integration process
* integration length
* desmearing (would be used to adjust noise threshold appropriately)
* range searched

**Comment by n-west** [comment]
This is primarily to support #5 and I'm working on it as part of a bit of a refactor towards progress on #7.

I've got the following struct as a work-in-progress towards what to return from drift integrations rather than just the ndarray that is currently returned
```
struct frequency_drift_plane {
        struct drift_rates {
                int index_in_plane;
                double drift_rate_Hz_per_sec=0.0F;
                int desmeared_bins=1; // number of bins per spectra used to desmear
        };
        int64_t integration_steps;
        std::vector<drift_rates> drift_rate_info;
        bland::ndarray integrated_drifts;
};
```

The idea would be the `index_in_plane` makes sure if the `vector<drift_rates>` gets shuffled we can always know what index along drift axis this is relevant to, the `drift_rate_Hz_per_sec` tracks the physical units and the `desmeared_bins` is the number of bins per spectra we integrate when desmearing is on (this is the number of bins a signal would expect to cross. This would provide required info to fix #5 in the hit search

**Comment by n-west** [comment]
Working on this in the desmear-noise-comp branch, the final data structure attached is 
```
struct frequency_drift_plane {
    struct drift_rate {
            int index_in_plane;
            double drift_rate_slope = 0.0F;
            double drift_rate_Hz_per_sec = 0.0F;
            int desmeared_bins=1; // number of bins per spectra used to desmear
    };

    frequency_drift_plane(bland::ndarray drift_plane, integrated_flags drift_rfi) : _integrated_drifts(drift_plane), _dedrifted_rfi(drift_rfi) {}
    frequency_drift_plane(bland::ndarray drift_plane, integrated_flags drift_rfi, int64_t integration_steps, std::vector<drift_rate> dri) : 
        _integrated_drifts(drift_plane), _dedrifted_rfi(drift_rfi), _integration_steps(integration_steps), _drift_rate_info(dri) {
    }
    

    // slow-time steps passed through for a complete integration, the total number
    // of bins contributing to this integration is demsear_bins * integration_steps
    int64_t _integration_steps;

    std::vector<drift_rate> _drift_rate_info; // info for each drift rate searched (consider changing to map with key being integer number of unit drifts)

    // The actual frequency drift plane
    bland::ndarray _integrated_drifts;

    integrated_flags _dedrifted_rfi;

};
```

`coarse_channels` hold on to this rather than several other fields so this structure keeps them all together
